### PR TITLE
Dip0001-related adjustments (inc. constants)

### PR DESCRIPTION
**Pull request by the-schnibble**

### CODE REVIEW

File: src/validation.h [hunk 1]
```diff
@@ -73,11 +73,11 @@ static const unsigned int DEFAULT_DESCENDANT_SIZE_LIMIT = 101;
...
 static const unsigned int DEFAULT_MEMPOOL_EXPIRY = 72;
 /** The maximum size of a blk?????.dat file (since 0.8) */
-static const unsigned int MAX_BLOCKFILE_SIZE = 0x8000000; // 128 MiB
+static const unsigned int MAX_BLOCKFILE_SIZE = 0x10000000; // 256 MiB
 /** The pre-allocation chunk size for blk?????.dat files (since 0.8) */
-static const unsigned int BLOCKFILE_CHUNK_SIZE = 0x1000000; // 16 MiB
+static const unsigned int BLOCKFILE_CHUNK_SIZE = 0x2000000; // 32 MiB
 /** The pre-allocation chunk size for rev?????.dat files (since 0.8) */
-static const unsigned int UNDOFILE_CHUNK_SIZE = 0x100000; // 1 MiB
+static const unsigned int UNDOFILE_CHUNK_SIZE = 0x200000; // 2 MiB
```
Comment (UdjinM6) [inline src/validation.h]: Are these 3 above actually DIP0001-related? IMO storing blocks on disk is kind of perpendicular to consensus rules.

File: src/net.h [hunk 2]
```diff
@@ -56,7 +56,7 @@ static const unsigned int MAX_INV_SZ = 50000;
 /** The maximum number of new addresses to accumulate before announcing. */
 static const unsigned int MAX_ADDR_TO_SEND = 1000;
 /** Maximum length of incoming protocol messages (no message over 2 MiB is currently acceptable). */
-static const unsigned int MAX_PROTOCOL_MESSAGE_LENGTH = 2 * 1024 * 1024;
+static const unsigned int MAX_PROTOCOL_MESSAGE_LENGTH = 3 * 1024 * 1024;
```
Comment (UdjinM6) [inline src/net.h]: The comment above should be fixed as well.

File: src/validation.h (hunk 1 above)
Comment (the-schnibble) [reply src/validation.h]: Agree. We shouldn't scale storing blocks on disk. Thus `MIN_DISK_SPACE_FOR_BLOCK_FILES` should be 945 MB

File: src/net.h (hunk 2 above)
Comment (the-schnibble) [reply src/net.h]: fixed

File: src/validation.cpp [hunk 3]
```diff
@@ -4369,7 +4369,7 @@ std::string GetWarnings(const std::string& strFor)
 
 ThresholdState VersionBitsTipState(const Consensus::Params& params, Consensus::DeploymentPos pos)
 {
-    LOCK(cs_main);
+    AssertLockHeld(cs_main);
```
Comment (UdjinM6) [inline src/validation.cpp]: While I agree that `AssertLockHeld` should be enough here, it feels rather unrelated to DIP0001 specifically, so I think its better to keep it as is for now. Revert this pls.

Comment (the-schnibble) [reply src/validation.cpp]: fixed (moved this change to a separate PR)
